### PR TITLE
[Merged by Bors] - Dial cached enr's before making subnet discovery query

### LINKED ISSUE #1365: Try dial cached ENR's when requesting subnet queries
## Description

Discovery caches last seen ENR's. We should attempt to connect to these peers during the subnet query search to reduce the load on discovery and time to take a query search. 

This should not violate the peer count constraint, or reputation constraint on peers.

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Closes #1365 

## Proposed Changes

Dial peers in the `cached_enrs` who aren't connected, aren't banned and satisfy the subnet predicate before making a subnet discovery query.

### CODE REVIEW

**Comment by pawanjay176** [comment]
Blocked on #1415

**Comment by AgeManning** [comment]
I think this is unblocked. Let me know when ready to review :)

**Comment by pawanjay176** [comment]
Will merge in latest changes and then request review!

**Comment by pawanjay176** [comment]
@AgeManning running this on medalla. Added some extra logs to see if the code path gets triggered which we should probs downgrade before merging if all else is okay.

**Comment by pawanjay176** [comment]
Simulator is failing. Will have a look tomorrow.

**Comment by pawanjay176** [comment]
@AgeManning ready for review now. On medalla, my node always seems to have sufficient peers on every subnet so never has to dial any cached enrs, but I think this should work as required.

File: beacon_node/eth2_libp2p/src/peer_manager/mod.rs
Comment (AgeManning) [inline beacon_node/eth2_libp2p/src/peer_manager/mod.rs:255]: The count isn't going to change here. The peers will likey just shift to `dialing` state and won't be coutned. So this check seems like it will never hit.

Comment (AgeManning) [reply beacon_node/eth2_libp2p/src/peer_manager/mod.rs:255]: I think its fine to just to `dial_cached_enrs_in_subnet` (these are likely stale) and then send off the discovery query. 
It will get queued. If we manage to connect by the time the subnet query starts, it will do this check for us.

**Review by AgeManning — APPROVED** [review]
LGTM

**Pull request closed without merging**
